app/src/main/java/com/github/se/eventradar/viewmodel/qrCode/ScanTicketQrViewModel.kt

```diff
@@ -0,0 +1,182 @@
+package com.github.se.eventradar.viewmodel.qrCode
+
+import android.util.Log
+import androidx.compose.runtime.Composable
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.se.eventradar.model.Resource
+import com.github.se.eventradar.model.User
+import com.github.se.eventradar.model.event.Event
+import com.github.se.eventradar.model.event.EventUiState
+import com.github.se.eventradar.model.repository.event.IEventRepository
+import com.github.se.eventradar.model.repository.user.IUserRepository
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+@HiltViewModel(assistedFactory = ScanTicketQrViewModel.Factory::class)
+class ScanTicketQrViewModel
+@AssistedInject // Dependency injection
+constructor(
+    private val userRepository: IUserRepository,
+    private val eventRepository: IEventRepository,
+    val qrCodeAnalyser: QrCodeAnalyser,
```

Comment:
question: Why isn't this private either?

Reply:
needs to be accessed elsewhere